lib/vendors/axis-communications/sv_vendor_axis_communications.c

```diff
@@ -197,10 +246,22 @@ sv_vendor_axis_communications_set_attestation_report(signed_video_t *sv,
     // If |certificate_chain| already exists, return error.
     if (self->certificate_chain) return SV_NOT_SUPPORTED;
     // Allocate memory and copy to |self|.
-    self->certificate_chain = calloc(1, strlen(certificate_chain) + 1);
+    size_t certificate_chain_size = strlen(certificate_chain);
+    if (certificate_chain_size == 0) goto catch_error;
+
+    bool has_newline_at_end = false;
+    if (certificate_chain[certificate_chain_size - 1] == '\n') {
+      has_newline_at_end = true;
+    }
+    // Allocate memory for |certificate_chain| + null-terminated character and maybe extra '\n'.
+    self->certificate_chain = calloc(1, certificate_chain_size + (has_newline_at_end ? 1 : 2));
     allocated_certificate_chain = true;
     if (!self->certificate_chain) goto catch_error;
     strcpy(self->certificate_chain, certificate_chain);
+    if (!has_newline_at_end) {
+      strcpy(self->certificate_chain + certificate_chain_size, "\n");
```

Comment:
Something feels really wrong here. strcpy -> strcat ?

Reply:
It works and has been verified, but I agree strcat would have been nicer. I consider removing all newlines later and then this will be removed.